### PR TITLE
Add services to Renault integration

### PR DESCRIPTION
**Pull request by epenet**
## Proposed change

Add services to Renault integration, as a follow up to #14383

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/54820
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/renault.markdown [hunk 1]
```diff
@@ -25,3 +25,50 @@ You need two API keys: one for Gigya and one for Kamereon and they shouldn't be
 {% include integrations/config_flow.md %}
 
 All vehicles linked to the account should then get added as devices, with sensors added as linked entity.
+
+## Services
+
+### Service `renault.ac_start`
+
+Start A/C on the specified vehicle by `vin`.
```
Comment (MartinHjelmare) [inline source/_integrations/renault.markdown]: Please update the docs after the change of direction in the parent PR.

File: source/_integrations/renault.markdown [hunk 2]
```diff
@@ -25,3 +25,41 @@ You need two API keys: one for Gigya and one for Kamereon and they shouldn't be
...
+  | `vehicle`| yes | device_id of the vehicle |
+
+### Service `renault.charge_set_schedules`
+
+Update charge schedule on vehicle.
+
+  | Service data attribute | Required | Description |
+  | ---------------------- | -------- | ----------- |
+  | `vehicle`| yes | device_id of the vehicle |
+  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules |
```
Comment (MartinHjelmare) [inline source/_integrations/renault.markdown]: The schedules parameter needs to be described in more detail.

File: source/_integrations/renault.markdown [hunk 3]
```diff
@@ -25,3 +25,41 @@ You need two API keys: one for Gigya and one for Kamereon and they shouldn't be
...
+
+### Service `renault.ac_start`
+
+Start A/C on vehicle.
+
+  | Service data attribute | Required | Description |
+  | ---------------------- | -------- | ----------- |
+  | `vehicle`| yes | device_id of the vehicle |
+  | `temperature` | yes | Target A/C temperature in °C |
+  | `when` | no | Timestamp for the start of the A/C (optional - defaults to now) |
```
Comment (MartinHjelmare) [inline source/_integrations/renault.markdown]: Maybe give an example of the timestamp format?

File: source/_integrations/renault.markdown [hunk 4]
```diff
@@ -50,10 +50,17 @@ Cancel A/C on vehicle.
...
   | `vehicle`| yes | device_id of the vehicle |
-  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules |
+  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules | `[{'id':1,'activated':true,'monday':{'startTime':'T12:00Z','duration':15}},<br>{'id':2,'activated':false,'monday':{'startTime':'T12:00Z','duration':240}}<br>]` |
+  
+Notes:
+
+- `schedules` can be in the form `{'id':1,...}` when updating a single schedules, or in the form `[{'id':1,...},{'id':2,...},...]` when updating multiple schedules within the same call
+- the `id` is compulsory on each `schedule` (should be 1 to 5 depending on the vehicle)
+- the `activated` flag is an optional boolean. If it is not provided, then the existing flag will be kept as is.
+- the `monday` to `sunday` elements are optional. If they are not provided, then the existing settings will be kept for each day. If they are provided as None, then the existing setting will be cleared. If a value is provided, it must conform to this format `{'startTime':'T12:00Z','duration':15}` where start time is in UTC format and the duration is in minutes.
```
Comment (MartinHjelmare) [inline source/_integrations/renault.markdown:63]: I noticed now that we only validate the time string as string in the service schema. Could we use the time validator to help users use the correct time string?

https://github.com/home-assistant/core/blob/33fb080c1e7dc5db8e24cc142fd9b1972ef337f3/homeassistant/helpers/config_validation.py#L351-L364

Then we can format the final string from the returned datetime.

Comment (epenet) [reply source/_integrations/renault.markdown:63]: The CLI version of RenaultAPI has some validation/conversion built in: https://github.com/hacf-fr/renault-api/blob/7c2336c52ed0298a3ed2b8e7dbb77b839676dced/src/renault_api/cli/charge/schedule.py#L202

There is also a draft PR to add validation to the non-CLI version: https://github.com/hacf-fr/renault-api/pull/157

There are some extra rules like the minutes of startTime needs to be in (00,15,30,45) and the total duration needs to be a multiple of 15 minutes (eg. 240 is valid but not 245)
There are also some overlap rules (Monday 23:45 for 120 minutes and Tuesday 00:15 for 30 minutes is invalid)

I'll have a go at a follow-up PR.

Comment (epenet) [reply source/_integrations/renault.markdown:63]: I'm not sure when I'll get a chance to work on the new PR (there is also a request from @frenck to implement re-auth).
Should this documentation PR be merged anyway in the meantime?

Comment (MartinHjelmare) [reply source/_integrations/renault.markdown:63]: Sure! The docs are fine.

**Review by MartinHjelmare — DISMISSED** [review]

File: source/_integrations/renault.markdown [hunk 5]
```diff
@@ -25,3 +25,48 @@ You need two API keys: one for Gigya and one for Kamereon and they shouldn't be
...
+  | `vehicle`| yes | device_id of the vehicle |
+
+### Service `renault.charge_set_schedules`
+
+Update charge schedule on vehicle.
+
+  | Service data attribute | Required | Description | Example |
+  | ---------------------- | -------- | ----------- | ------- |
+  | `vehicle`| yes | device_id of the vehicle |
+  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules | `[{'id':1,'activated':true,'monday':{'startTime':'T12:00Z','duration':15}},<br>{'id':2,'activated':false,'monday':{'startTime':'T12:00Z','duration':240}}<br>]` |
```
Comment (MartinHjelmare) [inline source/_integrations/renault.markdown]: I noticed in the preview that the br tag doesn't work.

https://deploy-preview-18981--home-assistant-docs.netlify.app/integrations/renault/

Comment (epenet) [reply source/_integrations/renault.markdown]: I'm not sure how to generate a preview, but I've moved the example out of the grid to gain some space.

Comment (MartinHjelmare) [reply source/_integrations/renault.markdown]: The preview is generated automatically by the CI build. It's linked below among the PR checks.

**Review by MartinHjelmare — APPROVED** [review]

**Comment by epenet** [comment]
Thanks

**Pull request merged**